src/parser/core/modules/Invulnerability.js

```diff
@@ -62,12 +62,18 @@ export default class Invulnerability extends Module {
 		for (let i = 0; i < events.length; i++) {
 			const event = events[i]
 
-			// Ignore non-damage events, and damage to friendlies
 			if (
-				!event.targetID ||
+				// Make sure it's got all the info we care about, and we're tracking it
 				!ACTIVITY_EVENT_TYPES.includes(event.type) ||
-				(event.type === 'damage' && event.targetIsFriendly) ||
-				(event.type === 'heal' && event.sourceIsFriendly)
+				!event.targetID ||
+				!event.sourceID ||
+				// we only care about casts and heals by players onto something other than themselves
+				(
+					['begincast', 'heal'].includes(event.type) &&
+					(!event.sourceIsFriendly || event.sourceID === event.targetID)
```

Comment:
Hm, on thinking about it a bit, this will possibly allow heals from healers onto current target to prevent downtime. Hold off merging, I'll run some tests in a bit.